pkg/util/provider/machinecontroller/ms_test.go

```diff
@@ -0,0 +1,133 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package controller
+
+import (
+	"context"
+
+	"github.com/gardener/machine-controller-manager/pkg/apis/machine/v1alpha1"
+	machinev1 "github.com/gardener/machine-controller-manager/pkg/apis/machine/v1alpha1"
+	"github.com/gardener/machine-controller-manager/pkg/util/provider/driver"
+	. "github.com/onsi/ginkgo"
+
+	. "github.com/onsi/gomega"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+var _ = Describe("machineCrashloopBackoff", func() {
+	objMeta := &metav1.ObjectMeta{
+		GenerateName: "class",
+		Namespace:    testNamespace,
+	}
+
+	// classKind := "MachineClass"
+	secretData := map[string][]byte{
+		"userData":            []byte("dummy-data"),
+		"azureClientId":       []byte("dummy-client-id"),
+		"azureClientSecret":   []byte("dummy-client-secret"),
+		"azureSubscriptionId": []byte("dummy-subcription-id"),
+		"azureTenantId":       []byte("dummy-tenant-id"),
+	}
+
+	Describe("machineCrashloopBackoff", func() {
+
+		It("Should delete the machine (old code)", func() {
+			stop := make(chan struct{})
+			defer close(stop)
+
+			// Create test secret and add it to controlCoreObject list
+			testSecret := &v1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-secret",
+					Namespace: testNamespace,
+				},
+				Data: secretData,
+			}
+
+			// Create a test secretReference because the method checkMachineClass needs it
+			testSecretReference := &v1.SecretReference{
+				Name:      "test-secret",
+				Namespace: testNamespace,
+			}
+
+			testMachineClass := &machinev1.MachineClass{
+				ObjectMeta: *newObjectMeta(objMeta, 0),
+				SecretRef:  testSecretReference,
+			}
+
+			controlCoreObjects := []runtime.Object{}
+			controlCoreObjects = append(controlCoreObjects, testSecret)
+
+			// Create test machine object in CrashloopBackoff state
+			testMachineObject1 := &machinev1.Machine{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "testmachine_1",
+					Namespace: testNamespace,
+				},
+				Status: machinev1.MachineStatus{
+					CurrentStatus: machinev1.CurrentStatus{
+						Phase: v1alpha1.MachineCrashLoopBackOff,
+					},
+				},
+			}
+
+			// Create another test machine object in Running state
+			testMachineObject2 := &machinev1.Machine{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "testmachine_2",
+					Namespace: testNamespace,
+				},
+				Status: machinev1.MachineStatus{
+					CurrentStatus: machinev1.CurrentStatus{
+						Phase: v1alpha1.MachineRunning,
+					},
+				},
+			}
+
+			controlMachineObjects := []runtime.Object{}
+			controlMachineObjects = append(controlMachineObjects, testMachineObject1)
+			controlMachineObjects = append(controlMachineObjects, testMachineObject2)
+
+			fakeDriver := driver.NewFakeDriver(&driver.FakeDriver{
```

Comment:
Please move the instance construction inside the `NewFakeDriver` func. The func can take what parameters are required.

Reply:
Yes, Amshu. I got your point fully now. Changes on the way!